app/src/org/commcare/activities/LoginActivityUiController.java

```diff
@@ -221,9 +223,28 @@ public void refreshView() {
 
         // Decide whether or not to show the app selection spinner based upon # of usable apps
         ArrayList<ApplicationRecord> readyApps = MultipleAppsUtil.getUsableAppRecords();
+        if(ConnectManager.isConnectIdIntroduced()) {
```

Comment:
should abstract into another method - filterConnectApps

Reply:
[49aef9b](https://github.com/dimagi/commcare-android/pull/2724/commits/49aef9b1630835182867aa080277d058dbd245c5)